File: 🟢 Easy/1331 - Rank Transform of an Array/_1331RankTransformofanArray.js
/**
 * @param {number[]} arr
 * @return {number[]}
 */
var arrayRankTransform = function (arr) {
    const set = new Set(arr);
    const sortedArr = [...set].sort((a, b) => a - b);

    let map = new Map();
    let r = 1;

    for (let el of sortedArr) {
        map.set(el, r);
        r++;
    }


    for (let i = 0; i < arr.length; i++) {
        arr[i] = map.get(arr[i]);
    }

    return arr;
};

File: 🟢 Easy/1331 - Rank Transform of an Array/_1331RankTransformofanArray.java
class Solution {
    public int[] arrayRankTransform(int[] arr) {
        Set<Integer> set = new TreeSet<>();

        for (int el : arr) {
            set.add(el);
        }

        Map<Integer, Integer> map = new HashMap<>();
        int r = 1;
        for (int s : set) {
            map.put(s, r);
            r++;
        }

        for (int i = 0; i < arr.length; i++) {
            arr[i] = map.get(arr[i]);
        }

        return arr;
    }
}